lib/poseidon/cluster_metadata.rb

```diff
@@ -49,6 +49,8 @@ def broker(broker_id)
 
     # Return lead broker for topic and partition
     def lead_broker_for_partition(topic_name, partition)
+      raise ::Poseidon::Errors::UnknownTopicOrPartition unless @topic_metadata[topic_name]
```

Comment:
@GordonDiggs I'm reusing an existing error mesage which obvi conflates the two concepts. My changes do not affect what would happen in the case you're describing. IOW, whatever used to happen will continue to happen :wink: